src/IO/S3/URI.cpp

```diff
@@ -146,6 +146,12 @@ URI::URI(const std::string & uri_)
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Bucket or key name are invalid in S3 URI.");
 }
 
+void URI::addRegionToURI(const std::string &region)
+{
+    auto pos = endpoint.find("amazonaws.com");
```

Comment:
I think we might want to also check if pos is found and then proceed. There is a chance that you might not find what you're searching for in the endpoint, in which case pos will be empty. Maybe check `(pos != std::string::npos)` ?

Reply:
you are right, thanks